src/core/dbus.c

```diff
@@ -1044,6 +1050,8 @@ void bus_done_private(Manager *m) {
                 destroy_bus(m, &b);
 
         m->private_buses = set_free(m->private_buses);
+        m->private_buses_subscribed = set_free(m->private_buses_subscribed);
```

Comment:
destroy_bus() should drop the private bus connection from the set, and please check if the set is empty before free.